### DISCUSSION TITLE
OAuth flow. Receiving invalid flow state, no valid flow state found

### DISCUSSION THREAD

**Discussion by haexhub** [supabase/supabase#19795] [Questions]
Hi everyone.

I'm trying to make supabase work with Oauth (github), but no matter what I'm doing, I always get and 403 with invalid flow state, no valid flow state found error.
I'm able to start the auth flow and come to the github sign in page. But after I've logged in myself and get to my redirect url I get this error.

![grafik](https://github.com/supabase/supabase/assets/29780781/a0c6e11a-a9bb-4a2c-8924-060b134d8720)

Steps to reproduce:

```
git clone https://github.com/supabase/supabase.git
```

add the env variables in docker-compose.yml gotrue service part

```
GOTRUE_EXTERNAL_GITHUB_ENABLED:  true
GOTRUE_EXTERNAL_GITHUB_CLIENT_ID: your_client_id
GOTRUE_EXTERNAL_GITHUB_SECRET:  your_client_secret
GOTRUE_EXTERNAL_GITHUB_REDIRECT_URI:  http://localhost:3000/auth/confirm # for the nuxt auth flow
```

```
git clone https://github.com/supabase-community/supabase-by-example.git
```

goto oauth-flow and choose any framework, I tried all of them. Adjust the env file accordingly and run the app.

Try the github auth flow and you will receive the mentioned error. (Or at least I do)

Could anybody give me the right hint, what I'm missing here?

What is also confusing me, if  I go to the dashboard, I can see that

![grafik](https://github.com/supabase/supabase/assets/29780781/c378a86c-d3fc-4d59-b988-dd8095cb4d7a)

There the github provider is not active. But it must be configured correctly, otherwise I would not be able to start the auth flow at all. So I think this overview is just meant for the cloud environment I guess?!
Guess it would be nice, if we could use this page with self hosted supabase as well.

**Comment by kangmingtay** [comment]
@haexhub what client libraries are you using for your proejct? are you using supabase-js or supabase/ssr?

**Reply by haexhub** [reply]
I'm using the supabase-js client, like its used in the ```https://github.com/supabase-community/supabase-by-example.git```
